eb_hooks.py

```diff
@@ -353,6 +357,28 @@ def parse_hook_CP2K_remove_deps_for_aarch64(ec, *args, **kwargs):
         raise EasyBuildError("CP2K-specific hook triggered for non-CP2K easyconfig?!")
 
 
+def parse_hook_zen4_module_only(ec, eprefix):
+    """
+    Use --force --module-only if building a foss-2022b-based EasyConfig for Zen4.
+    This toolchain will not be supported on Zen4, so we will generate a modulefile
+    and have it print an LmodError.
+    """
+    ecname, ecversion = ec['name'], ec['version']
+    tcname, tcversion = ec['toolchain']['name'], ec['toolchain']['version']
+    if (
+        (tcname in ['foss', 'gompi'] and tcversion == '2022b') or
+        (tcname in ['GCC', 'GCCcore'] and LooseVersion(tcversion) == LooseVersion('12.2.0')) or
+        (ecname in ['foss', 'gompi'] and ecversion == '2022b') or
+        (ecname in ['GCC', 'GCCcore'] and LooseVersion(ecversion) == LooseVersion('12.2.0'))
+    ):
+        update_build_option('force', 'True')
+        update_build_option('module_only', 'True')
+        # TODO: create a docs page to which we can refer for more info here
+        # TODO: then update the link to the known issues page to the _specific_ issue
+        errmsg = "Toolchains based on GCCcore-12.2.0 are not supported for the Zen4 architecture."
+        errmsg += " See https://www.eessi.io/docs/known_issues/eessi-2023.06/"
+        ec['modluafooter'] = 'LmodError(%s)' % errmsg
```

Comment:
You can an escape hatch by only triggering this if a certain environment variable is not set (some environment variable unique to the bot)